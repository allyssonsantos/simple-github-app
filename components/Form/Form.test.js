import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import 'jest-styled-components';

const getFormWithContext = context => {
  jest.doMock('../../providers/AppProvider', () => {
    return {
      AppConsumer: props => props.children(context),
    };
  });

  return require('./index').default;
};

describe('Form component', () => {
  const context = { searchUser: jest.fn() };

  describe('Snapshots', () => {
    it('Should match snapshot', () => {
      const Form = getFormWithContext(context);
      const component = renderer.create(<Form />);
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Context functions', () => {
    it('Should call searchUser from context', () => {
      const Form = getFormWithContext(context);
      const wrapper = mount(<Form />);

      expect(context.searchUser).toBeCalledTimes(0);

      wrapper.find('form').simulate('submit');

      expect(context.searchUser).toBeCalledTimes(1);
    });
  });

  describe('Change functions', () => {
    it('Should call onChange from input', () => {
      const Form = getFormWithContext(context);
      const wrapper = mount(<Form />);
      const newValue = 'new value';

      wrapper.find('input').simulate('change', { target: { value: newValue } });

      expect(wrapper.state().value).toBe(newValue);
    });
  });
});
