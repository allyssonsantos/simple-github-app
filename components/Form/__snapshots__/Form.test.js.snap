// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component Snapshots Should match snapshot 1`] = `
.c0 {
  border: 1px solid #999;
  border-radius: 22px;
  font-size: 18px;
  height: 42px;
  padding: 10px;
}

<form
  onSubmit={[Function]}
>
  <input
    className="c0"
    name="search"
    onChange={[Function]}
    placeholder="Insert github username"
    type="text"
    value=""
  />
</form>
`;
